### PR TITLE
Edit code on right-click

### PR DESCRIPTION
**Pull request by antoine-levitt**
This calls edit() to look at the code on a right-click in the GTK backend.

### CODE REVIEW

**Comment by timholy** [comment]
:+1: But I wonder if there's sometimes a reason to print without opening it in an editor. Should we check for a modifier key, too?

**Comment by antoine-levitt** [comment]
When would it be useful to print without opening the editor? It's the same information that's available when hovering. Maybe what would be useful is printing the line, maybe with context, ala grep -C.

**Comment by timholy** [comment]
I sometimes have so many stages of nested inlines, it's a little hard to parse the hover output, so I click to get a longer dump of that stage of the stack trace. Sometimes I also click on several bars, all of which are in the same file, and while I'd love to open an editor on the first click I probably don't need 3 more windows for 3 different clicks. That said, if I close the editor each time it would be fine.

In other words, I think this is a step forward and should probably be the default, but I wonder if we should have some way of avoiding the call to `edit` if, say, the user is holding down SHIFT or something when clicking.

**Comment by antoine-levitt** [comment]
I'm reluctant to use modifiers because they hurt discoverability. How about left click print info, right click edit?

**Comment by timholy** [comment]
Left-click is currently used to initiate a pan-drag operation. See https://juliagizmos.github.io/GtkReactive.jl/stable/zoom_pan.html. If you didn't already know this, then it proves your point about "discoverability." :smile:

It might be possible to use left-click for both, I guess, because the user releases the button quickly without dragging the mouse. I haven't thought carefully about how one would implement that, though.

**Comment by antoine-levitt** [comment]
Left-click by itself doesn't do anything here, I have to maintain CTRL to zoom.

**Comment by timholy** [comment]
Correct. But once you've zoomed in, you can "pan" the view by click-drag.

**Comment by antoine-levitt** [comment]
OK, then if you're fine with left-click doing two things (printing the line and panning), I find the committed changes pretty usable.

**Comment by timholy** [comment]
Thanks for a very nice improvement!
